docs/Environment variables.md
# Environment variables

| Name                               | Description                                                                                                                                                                                             |
| :--------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| API_ROOT                           | The url for a back end server instance for the service                                                                                                                                                  |
| ARCHIVED_DOCUMENTS_BASE_URL        | The url for the archived documents store                                                                                                                                                                |
| ASSETS_HOST                        | Optional host for assets CDN, defaults to app’s host                                                                                                                                                    |
| BASIC_AUTH_USER                    | Optional. If running a server with OAUTH2_BYPASS_SSO=true then this can be used in conjunction with a password to setup basic HTTP auth to protect the server                                           |
| BASIC_AUTH_PASSWORD                | Used in conjunction with BASIC_AUTH_USER to setup security for the server                                                                                                                               |
| CI                                 | Set to true for UAT testing, otherwise ignore                                                                                                                                                           |
| DATA_HUB_BACKEND_ACCESS_KEY_ID     | Identifier of a Hawk key required to access the backend metadata endpoints                                                                                                                              |
| DATA_HUB_BACKEND_SECRET_ACCESS_KEY | Secret Hawk key required to access the backend metadata endpoints                                                                                                                                       |
| FEATURES_FOLDER                    | Lo cation of the Cucumber feature files                                                                                                                                                                 |
| GOOGLE_TAG_MANAGER_KEY             | The key needed to integrate with Google Tag Manager to track usage                                                                                                                                      |
| LOG_LEVEL                          | How much logging to output: defaults to `debug` in dev and `error` everywhere else                                                                                                                      |
| METADATA_TTL                       | How long to store dropdown data etc for, in seconds. Defaults to 15 minutes                                                                                                                             |
| NODE_ENV                           | Whether to run the app in dev mode. Set to `production` in production, otherwise don't set for dev behaviour                                                                                            |
| OMIS_ARCHIVED_DOCUMENTS_BASE_URL   | The base URL for the OMIS archived document store. Holds archived quotes and deliverables                                                                                                               |
| OAUTH2_TOKEN_FETCH_URL             | OAuth fetch token url                                                                                                                                                                                   |
| OAUTH2_USER_PROFILE_URL            | OAuth user profile url                                                                                                                                                                                  |
| OAUTH2_AUTH_URL                    | OAuth auth url                                                                                                                                                                                          |
| OAUTH2_CLIENT_ID                   | OAuth client id                                                                                                                                                                                         |
| OAUTH2_CLIENT_SECRET               | OAuth client secret                                                                                                                                                                                     |
| OAUTH2_REDIRECT_URL                | OAuth callback url                                                                                                                                                                                      |
| OAUTH2_BYPASS_SSO                  | If a developer wishes to bypass OAuth locally then set this to true - defaults to `false`                                                                                                               |
| OAUTH2_DEV_TOKEN                   | Token used for working with OAuth locally whilst developing. This token is also used with CircleCi for providing user with different permissions                                                        |
| POSTCODE_KEY                       | Part of the frontend looks up addresses for postcodes using [getaddress.io](https://getaddress.io/). Obtain a key for the service and set it here                                                       |
| PROJECT_PHASE                      | Which badge to display in header: 'Alpha', 'Beta' or 'Demo' - defaults to 'Beta' @TODO - remove when Demo site is decommissioned                                                                        |
| PROXY                              | URL of a proxy to use to contact the API through. Useful for debugging                                                                                                                                  |
| QA_HOST                            | URL of the app under test                                                                                                                                                                               |
| QA_SELENIUM_HOST                   | URL of the Selenium server                                                                                                                                                                              |
| QA_SELENIUM_PORT                   | Port to use for the Selenium server                                                                                                                                                                     |
| REDISTOGO_URL                      | For use with heroku (deprecated)                                                                                                                                                                        |
| REDIS_HOST                         | You need to run redis and provide the host name for it here unless you specify the entire url - defaults to 'redis'                                                                                     |
| REDIS_PORT                         | Redis port, defaults to `6379`                                                                                                                                                                          |
| REDIS_URL                          | A full length url to connect to redis                                                                                                                                                                   |
| REDIS_USE_TLS                      | Boolean - whether Redis is proxied through stunnel or not. Defaults to false                                                                                                                            |
| SENTRY_DSN                         | Sentry DSN (optional)                                                                                                                                                                                   |
| SESSION_SECRET                     | String to encrypt session data with                                                                                                                                                                     |
| SESSION_TTL                        | How long the user session lasts, in millis. Defaults to 2 hours                                                                                                                                         |
| TEAM_SITE_SHARE_POINT_URL          | Team site share point url                                                                                                                                                                               |
| WEBPACK_ENV                        | Optionally select the webpack configuration variation to use, the default will correctly pick a production or development config based on NODE_ENV. Valid values include `prod`, `develop` and `docker` |
| HELP_CENTRE_URL                    | Help centre url                                                                                                                                                                                         |
| HELP_CENTRE_ANNOUNCMENTS_URL       | Help centre announcements url                                                                                                                                                                           |
| HELP_CENTRE_API_FEED               | Help centre feed API url                                                                                                                                                                                |
| ZEN_TICKETS_URL                    | Zendesk endpoint used to create tickets                                                                                                                                                                 |
| ZEN_EMAIL                          | Email address of the Zendesk account used for API authentication                                                                                                                                        |
| ZEN_TOKEN                          | Zendesk token used for API authentication                                                                                                                                                               |


src/config.js
const path = require('path')

const port = process.env.PORT || 3000
const isDev = (process.env.NODE_ENV !== 'production')
const defaultLogLevel = isDev ? 'debug' : 'error'

module.exports = {
  root: path.normalize(`${__dirname}/..`),
  env: process.env.NODE_ENV,
  isDev,
  port,
  apiRoot: process.env.API_ROOT || 'http://localhost:8000',
  api: {
    authUrl: '/token/',
    clientId: process.env.API_CLIENT_ID,
    clientSecret: process.env.API_CLIENT_SECRET,
  },
  postcodeLookup: {
    apiKey: process.env.POSTCODE_KEY,
    baseUrl: 'https://api.getAddress.io/v2/uk/{postcode}?api-key={api-key}',
  },
  redis: {
    url: process.env.REDIS_URL || process.env.REDISTOGO_URL,
    port: process.env.REDIS_PORT || 6379,
    host: process.env.REDIS_HOST || 'redis',
    metadataTtl: (process.env.METADATA_TTL || (15 * 60)),
  },
  googleTagManagerKey: process.env.GOOGLE_TAG_MANAGER_KEY,
  session: {
    secret: process.env.SESSION_SECRET || 'howdoesyourgardengrow',
    // 2 hour timeout
    ttl: process.env.SESSION_TTL || (2 * 60 * 60 * 1000),
  },
  logLevel: process.env.LOG_LEVEL || defaultLogLevel,
  zenUrl: `https://${process.env.ZEN_DOMAIN}.zendesk.com/api/v2/tickets.json`,
  zenToken: process.env.ZEN_TOKEN,
  zenEmail: process.env.ZEN_EMAIL,
  zenBrowser: process.env.ZEN_BROWSER,
  zenImpact: process.env.ZEN_IMPACT,
  zenService: process.env.ZEN_SERVICE,
  longDateFormat: 'D MMMM YYYY',
  mediumDateFormat: 'D MMM YYYY',
}
